master/internal/rm/kubernetesrm/pods.go

```diff
@@ -334,6 +364,185 @@ func (p *pods) getSystemResourceRequests(ctx *actor.Context) error {
 	return nil
 }
 
+func (p *pods) reattachAllocationPods(ctx *actor.Context, msg reattachAllocationPods) error {
+	listOptions := metaV1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", determinedLabel, msg.allocationID),
+	}
+
+	pods, err := p.podInterface.List(context.TODO(), listOptions)
+	if err != nil {
+		return errors.Wrap(err, "error listing pods checking if they can be restored")
+	}
+
+	configMaps, err := p.configMapInterface.List(context.TODO(), listOptions)
```

Comment:
i think in the general case a pod can survive without its primary config map if it is already started (we mount it and immediately unpack it into the container), but this is still probably fine.